core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/descriptors/StreamletDescriptor.scala

```diff
@@ -58,17 +57,17 @@ object ConfigParameterDescriptor {
 
 object StreamletDescriptor extends DefaultJsonProtocol {
 
-  def jsonDescriptor(streamlet: Streamlet): String = {
-    streamlet.toJson.compactPrint
+  def jsonDescriptor[Context <: StreamletContext](streamlet: Streamlet[Context]): String = {
```

Comment:
I don't like the side effect of this on the code..

Reply:
This can be improvided